### ISSUE TITLE
Concerns regarding design of zip deployment support for app services

### ISSUE THREAD

**Issue by Stolpe** [terraform-providers/terraform-provider-azurerm#16949]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.1

### AzureRM Provider Version

3.8.0

### Affected Resource(s)/Data Source(s)

azurerm_[linux|windows]_web_app, azurerm_[linux|windows]_web_app_slot

### Terraform Configuration Files

```hcl
resource "azurerm_app_service_active_slot" "app_service_api_active_slot" {
  count = var.app_service_slot_swap ? 1 : 0

  resource_group_name   = azurerm_app_service_slot.app_service_api_staging_slot.resource_group_name
  app_service_name      = azurerm_app_service_slot.app_service_api_staging_slot.app_service_name
  app_service_slot_name = azurerm_app_service_slot.app_service_api_staging_slot.name
}
```

### Debug Output/Panic Output

```shell
Not applicable
```

### Expected Behaviour

I regularly check the change log of this provider, and today I noticed that `azurerm_[linux|windows]_web_app` and `azurerm_[linux|windows]_web_app_slot` just got support for zip deployments. This is cool and all, but I have some concerns about how it is modeled. I wrote this in a comment of the PR (after it was merged), so I am not sure that is will be noticed. Since I know that you often release new versions on Fridays, and there is a time zone difference between the US where most of you work, and Sweden where I live, I figured it was best to create an issue about this too.

This is my comment:
> Hi,
>
> I know that this is now merged and accepted, but would it not have been preferred to model this as a separate resource? In many use cases, one would probably want to perform some kind of configuration transformation on appsettings.json or web.config prior to deploying it, and this transformation is likely partly based on output from Terraform.
>
> Wouldn't it have been better to have modeled this similarly to eg. the azurerm_app_service_active_slot resource. I use that resource with an extra parameter, which activates the swap on a second run with the otherwise same variables, which triggers the swap. That would have been a viable solution here too, but now since it's a property, wouldn't that be a lot harder?

Basically, I question whether this feature should really be implemented as a property on the app service resource. As I see it, it would be a better fit to have this as a separate resource much like how slot swapping is modeled in the provider.

If there is something fundamental that I'm missing, and this design is indeed the best way to do it, go ahead with it. However, I would prefer if we don't add another one of those "use property or separate resource, but not both" scenarios that exist here and there, so maybe it would be better not to release this feature in this shape if you agree with me.

### References

https://github.com/hashicorp/terraform-provider-azurerm/pull/16779#issuecomment-1136898372
